patches/chrome-android-java-src-org-chromium-chrome-browser-sync-settings-ManageSyncSettings.java.patch

```diff
@@ -1,7 +1,16 @@
 diff --git a/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/ManageSyncSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/ManageSyncSettings.java
-index 3b229431fc69c56bcbd4163ae46285a489882bf2..ee4d68cbdf2f0b83a02bb11e0429cf03462fb046 100644
+index 3b229431fc69c56bcbd4163ae46285a489882bf2..028279ee2599af11a538234a5a28df0bddad02f2 100644
 --- a/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/ManageSyncSettings.java
 +++ b/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/ManageSyncSettings.java
+@@ -53,7 +53,7 @@ import org.chromium.chrome.browser.sync.ui.PassphraseDialogFragment;
+ import org.chromium.chrome.browser.sync.ui.PassphraseTypeDialogFragment;
+ import org.chromium.chrome.browser.ui.signin.SignOutDialogCoordinator;
+ import org.chromium.chrome.browser.ui.signin.SignOutDialogCoordinator.Listener;
+-import org.chromium.components.browser_ui.settings.ChromeBaseCheckBoxPreference;
++import org.chromium.components.browser_ui.settings.brave_tricks.checkbox_to_switch.ChromeBaseCheckBoxPreference;
```

Comment:
we can't do this with asm?

Reply:
change `ChromeBaseCheckBoxPreference` -> `org.chromium.components.browser_ui.settings.brave_tricks.checkbox_to_switch.ChromeBaseCheckBoxPreference`

Reply:
or better yet, use asm to change the superclass for `ChromeBaseCheckBoxPreference` so we don't need to patch any sources in either?

Reply:
cc: @samartnik @AlexeyBarabash

Reply:
We need to change it in this file only cc @AlexeyBarabash.
We can extend our bytecode patching to apply patch for certain classes only, but atm there is no such ability. I'll create a follow up issue to make it possible via bytecode.

Reply:
Created issue https://github.com/brave/brave-browser/issues/31048